packages/twenty-front/src/modules/activities/emails/components/MessageThread.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import { useRecoilState } from 'recoil';
+
+import { MessageThreadBody } from '@/activities/emails/components/MessageThreadBody';
+import { MessageThreadBodyPreview } from '@/activities/emails/components/MessageThreadBodyPreview';
+import { MessageThreadSender } from '@/activities/emails/components/MessageThreadSender';
+import { MockedEmailUser } from '@/activities/emails/mocks/mockedThreads';
+import { viewableMessageThreadIdsFamilyState } from '@/activities/emails/state/viewableMessageThreadIdsFamilyState';
+
+const StyledMessageThread = styled.div`
+  border-bottom: 1px solid ${({ theme }) => theme.border.color.light};
+  cursor: pointer;
+  display: flex;
+  flex-direction: column;
+  padding: 16px 24px;
+`;
+
+const StyledMessageThreadHeader = styled.div`
+  display: flex;
+  flex-direction: column;
+  padding-bottom: 8px;
+  justify-content: space-between;
+`;
+
+type MessageThreadProps = {
+  id: string;
+  body: string;
+  sentAt: string;
+  from: MockedEmailUser;
+  to: MockedEmailUser[];
+};
+
+export const MessageThread = ({
+  id,
+  body,
+  sentAt,
+  from,
+}: MessageThreadProps) => {
+  const { displayName, avatarUrl } = from;
+  const [openedMessageId, setIsOpenedMessage] = useRecoilState(
+    viewableMessageThreadIdsFamilyState(id),
+  );
+
+  const isOpenedMessage = openedMessageId === id;
+
+  const updateIsOpenedMessage = () => {
+    if (isOpenedMessage) {
+      setIsOpenedMessage(null);
+    } else {
+      setIsOpenedMessage(id);
+    }
+  };
```

Comment:
Should I build a hook useThreadMessage ? Or is it acceptable to keep that logic in the component?